tests/v3_api/test_workload.py

```diff
@@ -82,6 +82,86 @@ def test_wl_cronjob():
     validate_workload(p_client, workload, "cronJob", ns.name)
 
 
+def test_wl_upgrade():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    con = [{"name": "test1",
+           "image": "sangeetha/testclient"}]
+    name = random_test_name("default")
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id)
+    workload = wait_for_wl_to_active(p_client, workload)
+    con = [{"name": "test1",
+           "image": "nginx"}]
+
+    p_client.update(workload, containers=con)
+    wait_for_pods_in_workload(p_client, workload, 1)
+    p_client.action(workload, "rollback",revision=2)
+    wait_for_pods_in_workload(p_client, workload, 1)
+    p_client.list_workload(workload=workload.id)
+    validate_workload_image(p_client, workload, "sangeetha/testclient")
+
+
+def test_wl_pod_scale_up():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    con = [{"name": "test1",
+           "image": "sangeetha/testclient"}]
+    name = random_test_name("default")
+    workload = p_client.create_workload(name=name,
+                                        containers=con,
+                                        namespaceId=ns.id)
+    workload = wait_for_wl_to_active(p_client, workload)
+    p_client.update(workload, scale=2)
+    validate_workload(p_client, workload, "deployment", ns.name, 2)
```

Comment:
@adingilloRancher After scaling up workloads , make sure that existing pods are not getting recreated. Check that the pods from before upgrade continue to be in "running" state.